src/main/java/com/google/googleidentity/oauth2/authorizationcode/AuthorizationCodeService.java

```diff
@@ -72,21 +89,29 @@ public String getCodeForRequest(OAuth2Request request){
         return codeStore.consumeCode(code);
     }
 
-    private String generateCode(){
+    /**
+     * Associate with client and username to reduce collisions.
+     *
+     * @param clientID
+     * @param username
+     * @return
+     */
+    private String generateCode(String clientID, String username){
+        byte[] fixInfo =
+                Hashing.sha256()
+                        .hashString(clientID + username, Charsets.UTF_8).asBytes();
 
-        byte[] bytes = new byte[codeLength];
-        random.nextBytes(bytes);
+        byte[] bytes = new byte[byteLength];
```

Comment:
The prefix and random bytes generation are interleaved. It would be easier to read if you kept the prefix logic together.

// generate random bytes
// compute prefix and copy over random bytes
// encode and truncate auth code string